terrawrap/utils/config.py

```diff
@@ -86,6 +88,130 @@ def parse_wrapper_configs(wrapper_config_files: List[str]) -> WrapperConfig:
+    dependencies = []
+
+    for file_name in os.listdir(directory):
+        if file_name.endswith('.tf_wrapper'):
+            file_name = os.path.join(directory, file_name)
+            wrapper_config_file = file_name
+            with open(wrapper_config_file) as wrapper_config_file:
+                wrapper_config = yaml.safe_load(wrapper_config_file)
+
+            try:
+                wrapper_dependencies = wrapper_config.get('depends_on')
+                if wrapper_dependencies:
+                    for path in wrapper_dependencies:
+                        path = get_absolute_path(path)
+                        dependencies.append(path)
+                    return dependencies
+            except TypeError:
+                return dependencies
+    return dependencies
+
+
+def single_config_dependency_grapher(config_dir: str, graph: networkx.DiGraph, visited: List[str]):
+    """
+    Given a directory, recursively finds all other directories it depends on and builds a graph.
+    :param config_dir: The config directory to obtain a dependency graph for
+    :param graph: The graph to add dependency info to. Empty at first, reused in recursion.
+    :param visited: A list of visited nodes. Empty at first, reused in recursion
+    """
+    if config_dir in visited:
+        return
+    visited.append(config_dir)
+    if is_config(config_dir):
+        graph.add_node(config_dir)
+
+    tf_dependencies = parse_dependencies(config_dir)
+    for dependency in tf_dependencies:
+        graph.add_node(dependency)
+        if config_dir in graph:
+            graph.add_edge(dependency, config_dir)
+
+    wrappers = find_wrapper_config_files(config_dir)
+    wrappers.reverse()  # we want the closest wrapper file that gives inherited dependencies
+    for wrapper in wrappers:
+        wrapper_dir = os.path.dirname(wrapper)
+        if wrapper_dir == config_dir:
+            continue
+        if has_depends_on(wrapper_dir):
+            closest_inheritance = wrapper_dir
+            inherited_dependencies = parse_dependencies(closest_inheritance)
+            added = False
+            for dependency in inherited_dependencies:
+                if dependency == config_dir:
+                    continue
+                added = True
+                graph.add_node(dependency)
+                if config_dir in graph:
+                    graph.add_edge(dependency, config_dir)
+            if added:
+                break  # we only need the closest, the recursion will handle anything higher
+
+    for predecessor in list(graph.predecessors(config_dir)):
+        single_config_dependency_grapher(predecessor, graph, visited)
+
+
+def directory_dependency_grapher(starting_dir: str) -> Tuple[networkx.DiGraph, List[str]]:
+    """
+    Given a starting directory, walks it and returns all dependency info.
+    :param starting_dir: The starting directory
+    :return: directory_graph: A graph composed of all dependency information for a directory
+    """
```

Comment:
possibly loop through and create a map of directory to 
walk throught every dir with tf and figure out the wrapper config for that dir 
dict = {dir_path : wrapper config}
configure to take in the wrapper config object